src/Umbraco.Core/Services/ContentService.cs

```diff
@@ -1131,6 +1131,9 @@ public void MoveToRecycleBin(IContent content, int userId = 0)
                     return;
                 }
 
+                // Add a relation for the item being deleted, so that we can know the original parent for if we need to restore later
+                ApplicationContext.Current.Services.RelationService.Relate(content.Parent(), content, Constants.Conventions.Relations.RelateParentDocumentOnDelete);

```

Comment:
All makes sense. Also I've realised that I hadn't handled the scenario of emptying the recycle-bin - so the relation entries would have been orphaned. Now I can handle this in the custom event handler :+1: